### PR TITLE
Leaderboard update

### LINKED ISSUE #4: Scoreboard overlaps with courses
I imagine there will be UI changes eventually and this issue isn't really that big of a deal, since it doesn't affect the functioning of the app.

The scoreboard overlaps with the courses if the window is small enough, such as when my browser is taking up half of my laptop screen
window_bug

### LINKED ISSUE #6: Opt-Out of leaderboard
The approach I would take is:
- Add a setting to the settings page to opt out of the leaderboard
- Add a Settings model with both a user's id and their opt-out status (default opt-in) to keep track of user preferences
- when populating the 'top5' in the School model, query the Settings model to see if any of the top have opted-out. If so then do not put them in the list of the top5.

### LINKED ISSUE #21: Better UI/UX
Looking for specific feedback to help with this. 
My UI/UX skills are not great so any input would be helpful.

If you are a designer or developer who wants to join the team please email me at vaa6dx@virginia.edu or send in a pull request to the README

### PR DESCRIPTION
**Pull request by verndrade**
Adds opt-out and makes leaderboard visible again

TODO: Fix UI (make position not fixed), Filter hidden syllabi?

### CODE REVIEW

**Comment by winstliu** [comment]
`self.userprofile_set.annotate(submissions=Count('user__submission', filter=Q(user__submission__hidden=False))).filter(submissions__gt=0).order_by('-submissions')[:5]`

File: templates/settings.html [hunk 1]
```diff
@@ -11,7 +25,7 @@ <h4>Delete account</h4>
   <div class="form-group">
     <label for="suggestion">Username</label>
     <input class="form-control" id="suggestion" type="text" name="username" required>
-    <small class="form-text{# text-muted #}">Enter your username ({{ user.username }}) to confirm account deletion</small>
+    <small class="form-text{# text-muted #}">Enter your email ({{ user.username }}) to confirm account deletion</small>
```
Comment (winstliu) [inline templates/settings.html]:
```suggestion
    <small class="form-text{# text-muted #}">Enter your email ({{ user.email }}) to confirm account deletion</small>
```

File: templates/settings.html [hunk 2]
```diff
@@ -11,7 +25,7 @@ <h4>Delete account</h4>
   <div class="form-group">
     <label for="suggestion">Username</label>
```
Comment (winstliu) [inline templates/settings.html]:
```suggestion
    <label for="suggestion">Email</label>
```

File: templates/settings.html [hunk 3]
```diff
@@ -1,8 +1,22 @@
...
 
-<h3>Settings</h3>
+<h3 class="text-center">Settings</h3>
+
+
+<h4>Leaderboard</h4>
+<form method="post">
+  {% csrf_token %}
+  {% if user.profile.hide_name %}
+    <p>You are currently Anonymous</p>
```
Comment (winstliu) [inline templates/settings.html]:
```suggestion
    <p>Your username is hidden from other users</p>
```

File: templates/settings.html [hunk 4]
```diff
@@ -1,8 +1,22 @@
...
+    <input class="btn btn-success" type="submit" name="opt" value="Show Username">
+  {% else %}
+    <p>You are currently displaying your username: {{user.username | upToCharacter:"@"}}</p>
```
Comment (winstliu) [inline templates/settings.html]:
```suggestion
    <p>Your username is visible to other users</p>
```

File: templates/settings.html [hunk 5]
```diff
@@ -1,8 +1,22 @@
...
-<h3>Settings</h3>
+<h3 class="text-center">Settings</h3>
+
+
+<h4>Leaderboard</h4>
+<form method="post">
+  {% csrf_token %}
+  {% if user.profile.hide_name %}
+    <p>You are currently Anonymous</p>
+    <input class="btn btn-success" type="submit" name="opt" value="Show Username">
```
Comment (winstliu) [inline templates/settings.html]: Lowercase username here, same as below

**Review by winstliu — APPROVED** [review]
This is excellent, love the new styling. Only thing I would change is adding a bit more margin when the leaderboard moves below the courses. It looks a little cramped:
minimal-margin

**Comment by mengsarah** [comment]
> This is excellent, love the new styling. Only thing I would change is adding a bit more margin when the leaderboard moves below the courses.

I was going to say that I just changed that but oops, for about five minutes I forgot that pull requests and branches are two different things. It's tiny enough I feel like I shouldn't open an entirely new PR though haha.

I went into `index.html` and added the `mb-3` class to the div that holds `<h4>There {{num|pluralize:"is,are"}} {{num}} syllab{{num|pluralize:"us,i"}} for the {{school}}!</h4>`. Adds a bit of extra space to the bottom on large enough screens, too, but I think it still looks okay.

**Pull request merged**